### PR TITLE
feat(storage): map settlement jobs by certificate id

### LINKED ISSUE #1430: Column for certificate id -> settlement job id storage, with store/read helpers

### PR DESCRIPTION
**Pull request by Freyskeyd**
This adds a certificate-keyed index for settlement jobs in storage so
callers can resolve settlement data directly from a certificate id.

- add settlement_job_id_per_certificate column family, schema types,
  and registration
- add insert/get APIs for certificate-to-job mapping and direct job
  lookup by certificate id
- add codec and state-store tests for success, duplicate inserts, and
  missing-parent cases

Refs #1430

### CODE REVIEW

File: crates/agglayer-storage/src/stores/state/settlement/mod.rs [hunk 1]
```diff
@@ -93,6 +122,36 @@ impl SettlementReader for StateStore {
...
+    fn insert_settlement_job_id_for_certificate(
+        &self,
+        certificate_id: &CertificateId,
+        settlement_job_id: &Ulid,
+    ) -> Result<(), Error> {
+        if self
+            .db
+            .get::<SettlementJobIdPerCertificateColumn>(certificate_id)?
+            .is_some()
+        {
```
[the remark this answers is not in the document]
Comment (Freyskeyd) [reply crates/agglayer-storage/src/stores/state/settlement/mod.rs:110]: That's fine I guess, what do you think @Ekleog-Polygon  ?
